google/cloud/internal/http_payload.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/internal/http_payload.h"
+#include "google/cloud/internal/curl_impl.h"
+#include "google/cloud/log.h"
+#include "absl/strings/ascii.h"
+#include "absl/strings/str_split.h"
+#include <sstream>
+
+namespace google {
+namespace cloud {
+namespace rest_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+CurlHttpPayload::CurlHttpPayload(std::unique_ptr<CurlImpl> impl,
+                                 Options options)
+    : impl_(std::move(impl)), options_(std::move(options)) {}
+
+StatusOr<std::size_t> CurlHttpPayload::Read(absl::Span<char> buffer) {
+  return impl_->Read(buffer);
+}
+
+std::multimap<std::string, std::string> CurlHttpPayload::Trailers() const {
+  std::multimap<std::string, std::string> trailers;
+  auto headers = impl_->headers();
+  auto iter = std::find_if(headers.begin(), headers.end(),
+                           [](std::pair<std::string, std::string> p) {
+                             absl::AsciiStrToLower(&p.first);
+                             return p.first == "trailer";
```

Comment:
**not a full review, just a single comment:**

there is `absl::EqualsIgnoreCase()`. I'm guessing it's slightly faster than converting the whole string, then checking for equality. https://github.com/abseil/abseil-cpp/blob/b2c96417bd5c2b0a550611e503002a4594a932b2/absl/strings/match.h#L80-L81